### PR TITLE
test(FR-2311): expand diagnostics rule tests and extract parseStorageWarningThreshold

### LINKED ISSUE #6021: Expand diagnostics rule tests and extract parseStorageWarningThreshold
The diagnostics rule test suite has coverage gaps, particularly in storageProxyRules (only 6 tests) and endpointRules CORS edge cases. Additionally, parseStorageWarningThreshold is an untested utility function embedded in a hook.

## Why This Work is Needed

The diagnostics feature (FR-2300) has 130+ tests but storageProxyRules only has 6 tests (~70% coverage) and endpointRules CORS tests lack edge case coverage. parseStorageWarningThreshold is a pure function that should be extracted and independently tested.

## Expected Outcomes

- storageProxyRules: 6 → 14 tests (edge cases for negative capacity, overflow, rounding, large values)
- endpointRules: 11 → 22 tests (CORS edge cases, i18n key validation, error fallbacks)
- parseStorageWarningThreshold: 0 → 16 tests (extracted from hook, boundary/type validation)
- Total diagnostics tests: 130 → 183

JIRA Issue: FR-2311

### PR DESCRIPTION
**Pull request by agatha197**
Resolves #6021 ([FR-2311](https://lablup.atlassian.net/browse/FR-2311))

## Summary
- Expand `storageProxyRules` tests from 6 → 14 (edge cases: negative capacity, overflow, rounding, large values)
- Expand `endpointRules` tests from 11 → 22 (CORS edge cases, i18n key validation, error fallbacks)
- Extract `parseStorageWarningThreshold` from `useStorageProxyDiagnostics` hook to `storageProxyRules.ts` and add 16 tests
- Total diagnostics rule tests: 130 → 183

[FR-2311]: https://lablup.atlassian.net/browse/FR-2311?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: react/src/diagnostics/rules/__tests__/endpointRules.test.ts [hunk 1]
```diff
@@ -27,12 +27,55 @@ describe('checkEndpointReachability', () => {
...
+
+  it('should preserve empty string error message as fallback', () => {
+    const result = checkEndpointReachability(
+      'https://api.example.com',
+      false,
+      '',
+    );
+    // empty string is falsy, so falls back to 'Unknown error'
+    expect(result?.interpolationValues?.error).toBe('Unknown error');
+  });
```
[the remark this answers is not in the document]
Comment (agatha197) [reply react/src/diagnostics/rules/__tests__/endpointRules.test.ts]: Good catch. Renamed to 'should fall back to default error when empty string is provided' to match the actual assertion.

**Review by nowgnuesLee — APPROVED** [review]

File: react/src/diagnostics/rules/__tests__/endpointRules.test.ts [hunk 2]
```diff
@@ -27,12 +27,54 @@ describe('checkEndpointReachability', () => {
...
+  it('should include endpoint in interpolation values', () => {
+    const result = checkEndpointReachability(
+      'https://my-backend.example.com:8090',
+      false,
+    );
+    expect(result?.interpolationValues?.endpoint).toBe(
+      'https://my-backend.example.com:8090',
+    );
+  });
+
```
[the remark this answers is not in the document]
Comment (agatha197) [reply react/src/diagnostics/rules/__tests__/endpointRules.test.ts:58]: Good point. Removed the redundant test — the interpolation value is already verified in the 'should return critical when endpoint is unreachable' test.

File: react/src/diagnostics/rules/__tests__/endpointRules.test.ts [hunk 3]
```diff
@@ -27,12 +27,54 @@ describe('checkEndpointReachability', () => {
...
+  it('should use default error message for undefined errorMessage', () => {
+    const result = checkEndpointReachability(
+      'https://api.example.com',
+      false,
+      undefined,
+    );
+    expect(result?.interpolationValues?.error).toBe('Unknown error');
+  });
+
```
[the remark this answers is not in the document]
Comment (agatha197) [reply react/src/diagnostics/rules/__tests__/endpointRules.test.ts]: Agreed. Removed this redundant test — the undefined case is already covered by 'should use default error message when none provided'.
